src/platform/site-wide/user-nav/components/SearchMenu.jsx

```diff
@@ -1,65 +1,204 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { connect } from 'react-redux';
+import { toggleValues } from 'platform/site-wide/feature-toggles/selectors';
+import FEATURE_FLAG_NAMES from 'platform/utilities/feature-toggles/featureFlagNames';
 import classNames from 'classnames';
 import recordEvent from 'platform/monitoring/record-event';
+import debounce from 'platform/utilities/data/debounce';
 
 import { replaceWithStagingDomain } from '../../../utilities/environment/stagingDomains';
 import IconSearch from '@department-of-veterans-affairs/formation-react/IconSearch';
 import DropDownPanel from '@department-of-veterans-affairs/formation-react/DropDownPanel';
 
-import Typeahead, { isTypeaheadEnabled, typeaheadListId } from './Typeahead';
-
-class SearchMenu extends React.Component {
+export class SearchMenu extends React.Component {
   constructor(props) {
     super(props);
+    this.getSuggestions = debounce(
+      this.props.debounceRate,
+      this.getSuggestions,
+    );
     this.state = {
       searchAction: replaceWithStagingDomain('https://www.va.gov/search/'),
       userInput: '',
+      suggestions: [],
     };
   }
 
-  componentDidUpdate() {
-    this.refs.searchField.focus();
+  componentDidMount = () => {
+    if (this.props.searchTypeaheadEnabled) {
+      recordEvent({
+        event: 'search-type-ahead-enabled',
+      });
+      return;
+    }
+    recordEvent({
+      event: 'search-type-ahead-disabled',
+    });
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    const { userInput } = this.state;
+
+    const inputChanged = prevState.userInput !== userInput;
+    if (inputChanged) {
+      this.getSuggestions();
+    }
+  }
+
+  async getSuggestions() {
+    const { userInput } = this.state;
+
+    if (userInput?.length <= 2) {
+      if (this.state.suggestions.length > 0) {
+        this.setState({ suggestions: [] });
+      }
+
+      return;
+    }
+
+    const encodedInput = encodeURIComponent(userInput);
+    const response = await fetch(
```

Comment:
Around the fetch and the `await response.json()` we should probably add a `try {} catch (error) {}` in case `fetch` or the parsing of JSON fails for whatever reason 🙂